mobile/src/utils/functions.ts

```diff
@@ -0,0 +1,247 @@
+      } else if (response?.data?.error === "error") {
+        return "error";
+      } else {
+        return "error";
+      }
+    }
+  } else {
+    return false;
+  }
+};
+export const checkStoredKeys = async (oid: string) => {
+  if (isNil(oid)) {
+    return false;
+  }
+
+  const publicKey = await SecureStore.getItemAsync(oid + "_publicKey");
+  const privateKey = await SecureStore.getItemAsync(oid + "_privateKey");
+  const otpSeed = await SecureStore.getItemAsync(oid + "otpSeed");
+
+  if (isNil(publicKey) || isNil(privateKey)) {
+    const keys = await renewKeys(oid);
+    // verify device
+    const pubKey = keys.pubKey;
+    const deviceRegistered = await registerDevice(pubKey, oid);
+    return deviceRegistered;
+  }
+  if (isNil(otpSeed)) {
+    const deviceRegistered = await registerDevice(publicKey, oid);
+    return deviceRegistered;
+  }
+};
+
+export const getOid = async (): Promise<string | false> => {
+  const oid = await SecureStore.getItemAsync("oid");
+  if (isNil(oid)) {
+    return false;
+  }
+  return oid;
+};
+
+export const removeAllStoredData = async (whocallthisfunc: string) => {
+  // check if there is an oid
+  const oid = await SecureStore.getItemAsync("oid");
+  await SecureStore.deleteItemAsync(oid + "_publicKey");
+  await SecureStore.deleteItemAsync(oid + "_privateKey");
+  await SecureStore.deleteItemAsync(oid + "_deviceRegistered");
+  await SecureStore.deleteItemAsync("oid");
+  await SecureStore.deleteItemAsync("email");
+  await SecureStore.deleteItemAsync("phoneNumber");
+  await SecureStore.deleteItemAsync(oid + "RegistrationCompleted");
+  await SecureStore.deleteItemAsync(oid + "deviceVerified");
+  console.log("All stored data removed. request from: ", whocallthisfunc);
+};
+
+export const removeStoredData = async (keys: string[]) => {
+  for (const key of keys) {
+    await SecureStore.deleteItemAsync(key);
+  }
+};
+
+export const removeStoredKeys = async (whocallthisfunc: string) => {
+  const oid = await SecureStore.getItemAsync("oid");
+  await SecureStore.deleteItemAsync(oid + "_publicKey");
+  await SecureStore.deleteItemAsync(oid + "_privateKey");
+  console.log("Stored keys removed. request from: ", whocallthisfunc);
+};
+
+// For development: console log all stored data in secure store
+export const getAllStoredData = async () => {
+  const oid = await SecureStore.getItemAsync("oid");
+  const publicKey = await SecureStore.getItemAsync(oid + "_publicKey");
+  const privateKey = await SecureStore.getItemAsync(oid + "_privateKey");
+  const deviceRegistered = await SecureStore.getItemAsync(
+    oid + "_deviceRegistered"
+  );
+  const email = await SecureStore.getItemAsync("email");
+  const phoneNumber = await SecureStore.getItemAsync("phoneNumber");
+  //const deviceVerified = await SecureStore.getItemAsync(oid + "deviceVerified");
+  const RegistrationCompleted = await SecureStore.getItemAsync(
+    oid + "RegistrationCompleted"
+  );
+  console.warn("oid: ", oid);
```

Comment:
Please remove all `console.warn` statements. Especially the ones printing sensitive information :)

Reply:
This is only for development.

Reply:
Cool, happy to keep it in then for better local testing